src/Propulsion/Internal.fs

```diff
@@ -133,8 +132,8 @@ type Sem(max) =
     member _.HasCapacity = inner.CurrentCount <> 0
     member _.State = struct(max - inner.CurrentCount, max)
     member _.Wait(ct : CancellationToken) = inner.WaitAsync(ct)
-    member x.WaitButRelease() = // see https://stackoverflow.com/questions/31621644/task-whenany-and-semaphoreslim-class/73197290?noredirect=1#comment129334330_73197290
-        inner.WaitAsync().ContinueWith((fun _ -> x.Release()), CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
+    member x.WaitButRelease(ct) = // see https://stackoverflow.com/questions/31621644/task-whenany-and-semaphoreslim-class/73197290?noredirect=1#comment129334330_73197290
+        inner.WaitAsync().ContinueWith((fun _ -> x.Release()), ct, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
```

Comment:
this is the one I'm least sure of, would be good to understand what this is trying to do 👀